internal/core/drand_daemon_control.go

```diff
@@ -264,3 +266,48 @@ func (dd *DrandDaemon) Stop(ctx context.Context) {
 func (dd *DrandDaemon) WaitExit() chan bool {
 	return dd.exitCh
 }
+
+func SelfSignKeys(l log.Logger, multibeaconFolder string) ([]string, []*key.Pair, error) {
+	stores, err := key.NewFileStores(multibeaconFolder)
+	if err != nil {
+		return nil, nil, fmt.Errorf("drand: err reading beacons database: %w", err)
+	}
+
+	l.Infow("Detected stores", "amount", len(stores))
+
+	keys := make([]*key.Pair, 0)
+	beaconIDs := make([]string, 0)
+	for beaconID, fs := range stores {
+		pair, err := fs.LoadKeyPair()
+		if err != nil {
+			return nil, nil, fmt.Errorf("beacon id [%s] - error loading private/public: %w", beaconID, err)
+		}
+
+		// migration path: if a group wasn't reshared in a long time, its secret share won't containt the scheme name
+		// but the group will.
+		group, err := fs.LoadGroup()
+		if err != nil {
+			l.Warnw("beacon id [%s] - could not load group, please report this: %w", beaconID, err)
+		}
+		// the actual migration path
+		if group != nil && group.Scheme != nil {
+			pair.Public.Scheme = group.Scheme
+		}
+
+		// we validate signature after switching schemes
+		if pair.Public.ValidSignature() == nil {
+			l.Infow("beacon id [%s] - public identity already self signed.\n", beaconID)
+			continue
+		}
+
+		if err := pair.SelfSign(); err != nil {
+			return nil, nil, fmt.Errorf("failed to self-sign keypair for beacon id [%s]: %w", beaconID, err)
+		}
+		if err := fs.SaveKeyPair(pair); err != nil {
+			return nil, nil, fmt.Errorf("beacon id [%s] - saving identity: %w", beaconID, err)
+		}
+		keys = append(keys, pair)
+		beaconIDs = append(beaconIDs, beaconID)
+	}
+	return beaconIDs, keys, nil
```

Comment:
Not sure we really need to return all of these. Just returning an error should be fine, no?